examples/c/example.c

```diff
@@ -229,7 +228,43 @@ AranyaError run(Team *t) {
     EXPECT("error initializing team", err);
 
     // add sync peers.
-    err = add_sync_peers(t);
+
+    // Allocate memory for the config builder struct
+    struct AranyaSyncPeerConfigBuilder *builder = (struct AranyaSyncPeerConfigBuilder *)malloc(sizeof(struct AranyaSyncPeerConfigBuilder));
```

Comment:
We should let the C API provide a method for allocating/deallocating types defined in the API rather than manually invoking `malloc`. E.g. `new()` and `cleanup()` methods.

Reply:
Or define the type without using malloc like this:
```suggestion
    AranyaSyncPeerConfigBuilder builder;
```

Reply:
Yeah these should be stack allocated.

Reply:
Good call

Reply:
> We should let the C API provide a method for allocating/deallocating types defined in the API rather than manually invoking `malloc`. E.g. `new()` and `cleanup()` methods.

All types in our APIs are either

a. Completely defined and initialized via `Foo_init(Foo* foo)` methods
b. Forward declared and constructed via `Foo* Foo_new()` methods.

Providing `Foo* Foo_new()` constructors for completely defined types is a mistake. It means users can accidentally write code like this:

```c
Foo* foo = malloc(sizeof(Foo)); // oops, should have used `Foo_new`
[...]
Foo_free(foo); // oops, undefined behavior
```

All or almost all of our types use option (a) so that users can allocate our types however they please—stack, heap, or static memory. So, @ssweetney-spok's original code was correct (although I agree allocating it on the stack is simpler and better for this example).